containers/EmpSponsors.ts

```diff
@@ -0,0 +1,81 @@
+import { createContainer } from "unstated-next";
+import { useState, useEffect } from "react";
+import { BigNumber, Event } from "ethers";
+
+import Connection from "./Connection";
+import EmpContract from "./EmpContract";
+
+interface PositionState {
+  tokensOutstanding: BigNumber | null;
+  lockedCollateral: BigNumber | null;
+}
+
+interface SponsorDictionary {
+  [address: string]: PositionState;
+}
+
+interface EmpDictionary {
+  [address: string]: SponsorDictionary;
+}
+
+const useEmpSponsors = () => {
```

Comment:
This container fetches data for all sponsors that have ever created a position, and currently stores a list of all active sponsors with > 0 locked collateral